Sources/StreamChat/Workers/UserListUpdater.swift

```diff
@@ -63,3 +66,61 @@ enum UpdatePolicy {
     /// The resulting user set of the query will replace the existing user set.
     case replace
 }
+
+@available(iOS 13.0, *)
+extension UserListUpdater {
+    @discardableResult func update(userListQuery: UserListQuery, policy: UpdatePolicy = .merge) async throws -> [ChatUser] {
+        try await withCheckedThrowingContinuation { continuation in
+            update(userListQuery: userListQuery, policy: policy) { result in
+                continuation.resume(with: result)
+            }
+        }
+    }
+    
+    func fetch(userListQuery: UserListQuery, pagination: Pagination) async throws -> [ChatUser] {
+        let payload = try await withCheckedThrowingContinuation { continuation in
+            fetch(userListQuery: userListQuery.withPagination(pagination)) { result in
+                continuation.resume(with: result)
+            }
+        }
+        return payload.users.map { $0.asModel() }
+    }
+    
+    func loadUsers(_ userListQuery: UserListQuery, pagination: Pagination) async throws -> [ChatUser] {
+        let policy: UpdatePolicy = pagination.offset == 0 && pagination.cursor == nil ? .replace : .merge
+        return try await update(userListQuery: userListQuery.withPagination(pagination), policy: policy)
+    }
+    
+    func loadNextUsers(_ userListQuery: UserListQuery, limit: Int, offset: Int) async throws -> [ChatUser] {
+        try await loadUsers(userListQuery, pagination: Pagination(pageSize: limit, offset: offset))
+    }
+}
+
+extension UserListQuery {
+    func withPagination(_ pagination: Pagination) -> Self {
+        var query = self
+        query.pagination = pagination
+        return query
+    }
+}
+
+private extension UserPayload {
+    func asModel() -> ChatUser {
+        ChatUser(
+            id: id,
+            name: name,
+            imageURL: imageURL,
+            isOnline: isOnline,
+            isBanned: isBanned,
+            isFlaggedByCurrentUser: false, // TODO: isFlaggedByCurrentUser is not set
```

Comment:
We should check this.

Reply:
That is problematic since the property is local only, server does not have that. Yep, needs to be reviewed again.

Reply:
Only way to fix it is to make additional DB reads when converting API response to the model object. Currently this particular `asModel()` function is used for user search where this property is not used.

Reply:
I think it's not relevant for `ChatUser`, only for the `CurrentUser`. Let's just remove the TODO.